jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -963,49 +1049,51 @@ protected void doStop() throws Exception
         _scontext.clearAttributes();
     }
 
-
     /* ------------------------------------------------------------ */
     public boolean checkVirtualHost(final Request baseRequest)
     {
+        boolean match = false;
+
         if (_vhosts != null && _vhosts.length > 0)
         {
             String vhost = normalizeHostname(baseRequest.getServerName());
+            String connectorName = baseRequest.getHttpChannel().getConnector().getName();
 
-            boolean match = false;
-            boolean connectorName = false;
-            boolean connectorMatch = false;
-
-            for (String contextVhost:_vhosts)
+            for (int i = 0; i < _vhosts.length; i++)
             {
-                if (contextVhost == null || contextVhost.length()==0)
+                String contextVhost = _vhosts[i];
+                String contextVConnector = _vconnectors[i];
+
+                boolean connectorMatch = contextVConnector == null || contextVConnector.equalsIgnoreCase(connectorName);
+
+                if (contextVhost == null)
+                {
+                    if (connectorMatch && contextVConnector != null)
+                        return true;
+                    else
+                        continue;
+                }
+                else if (!connectorMatch)
                     continue;
-                char c=contextVhost.charAt(0);
-                switch (c)
+
+                if (_vhostswildcard[i])
                 {
-                    case '*':
-                        if (contextVhost.startsWith("*."))
-                            // wildcard only at the beginning, and only for one additional subdomain level
-                            match = match || contextVhost.regionMatches(true,2,vhost,vhost.indexOf(".") + 1,contextVhost.length() - 2);
-                        break;
-                    case '@':
-                        connectorName=true;
-                        String name=baseRequest.getHttpChannel().getConnector().getName();
-                        boolean m=name!=null && contextVhost.length()==name.length()+1 && contextVhost.endsWith(name);
-                        match = match || m;
-                        connectorMatch = connectorMatch || m;
-                        break;
-                    default:
-                        match = match || contextVhost.equalsIgnoreCase(vhost);
+                    // wildcard only at the beginning, and only for one additional subdomain level
+                    int index = vhost.indexOf(".");
+                    if (index >= 0)
+                        match = vhost.substring(index).equalsIgnoreCase(contextVhost);
                 }
+                else
+                    match = vhost.equalsIgnoreCase(contextVhost);
 
+                if (match)
+                    break;
```

Comment:
may as well be `return true` and get rid of `match` variable?

Reply:
I was on the same line of thinking when adjusting with the warning change.  Had already fixed this before I saw your change.